src/traits.rs

```diff
@@ -119,6 +119,45 @@ pub trait Context {
         hostcalls::cancel_grpc_call(token_id)
     }
 
+    fn create_grpc_stream(
+        &self,
+        cluster_name: &str,
+        service_name: &str,
+        method_name: &str,
+        initial_metadata: Vec<(&str, &[u8])>,
+    ) -> Result<u32, Status> {
+        hostcalls::create_grpc_stream(cluster_name, service_name, method_name, initial_metadata)
+    }
+
+    fn grpc_stream_send(
+        &self,
+        token_id: u32,
+        message: Option<&[u8]>,
+        end_stream: bool,
+    ) -> Result<(), Status> {
+        hostcalls::grpc_stream_send(token_id, message, end_stream)
+    }
+
+    fn grpc_stream_close(&self, token_id: u32) -> Result<(), Status> {
+        hostcalls::grpc_stream_close(token_id)
+    }
+
+    fn on_grpc_stream_receive_initial_metadata(&mut self, _token_id: u32, _headers: u32) {}
+
+    fn on_grpc_stream_receive_trailing_metadata(&mut self, _token_id: u32, _trailers: u32) {}
+
+    fn on_grpc_stream_receive_body(&mut self, _token_id: u32, _response_size: usize) {}
+
+    fn on_grpc_stream_close(&mut self, _token_id: u32, _status_code: u32) {}
+
+    fn get_grpc_call_initial_metadata(&self) -> Vec<(String, Vec<u8>)> {
+        hostcalls::get_map_bytes(MapType::GrpcReceiveInitialMetadata).unwrap()
+    }
+
+    fn get_grpc_call_trailing_metadata(&self) -> Vec<(String, Vec<u8>)> {
+        hostcalls::get_map_bytes(MapType::GrpcReceiveTrailingMetadata).unwrap()
+    }
+
```

Comment:
Could you please re-order them in order of use, i.e.:
- `open_grpc_stream`
- `on_grpc_stream_initial_metadata`
- `get_grpc_stream_initial_metadata`
- `send_grpc_stream_message`
- `on_grpc_stream_message`
- `get_grpc_stream_message` (currently missing)
- `on_grpc_stream_trailing_metadata`
- `get_grpc_stream_trailing_metadata`
- `close_grpc_stream`
- `on_grpc_stream_close`

Reply:
done